app/uk/gov/hmrc/merchandiseinbaggage/controllers/RetrieveDeclarationController.scala

```diff
@@ -38,10 +38,12 @@ class RetrieveDeclarationController @Inject()(
   mibConnector: MibConnector,
   view: RetrieveDeclarationView
 )(implicit appConfig: AppConfig, val ec: ExecutionContext)
-    extends DeclarationJourneyUpdateController {
+    extends DeclarationJourneyUpdateController with AmendDeclarationConfiguration {
 
   override val onPageLoad: Action[AnyContent] = actionProvider.journeyAction { implicit request =>
+    if(amendFlagConf.canBeAmended)
     Ok(view(form, routes.NewOrExistingController.onPageLoad(), request.declarationJourney.declarationType))
```

Comment:
i think `routes.NewOrExistingController.onPageLoad()` is a back link ?  if so this should use FF too ?